src/main/java/com/google/sps/DataServlet.java

```diff
@@ -153,8 +161,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     if (mutationNumber > currDataGraph.numMutations()) {
       diff = Utility.getMultiMutationAtIndex(mutList, mutationNumber);
     }
+    List<String> nodeNames = new ArrayList<>();
+    try {
+      JsonArray nodeNameArr = JsonParser.parseString(nodeNamesParam).getAsJsonArray();
+      for (int i = 0; i < nodeNameArr.size(); i++) {
+        String curr = nodeNameArr.get(i).getAsString().trim();
+        if (curr.length() > 0) {
+          nodeNames.add(curr);
+        }
+      }
+    } catch (JsonSyntaxException e) {
```

Comment:
I think the `getAsJsonArray` also throws an IllegalStateException which we used to handle?